packages/features/insights/components/RoutingFormResponsesTable.tsx

```diff
@@ -0,0 +1,371 @@
+  const { filter } = useFilterContext();
+  const tableContainerRef = useRef<HTMLDivElement>(null);
+  const { copyToClipboard, isCopied } = useCopy();
+
+  const {
+    dateRange,
+    selectedTeamId,
+    isAll,
+    initialConfig,
+    selectedRoutingFormId,
+    selectedMemberUserId,
+    selectedBookingStatus,
+    selectedRoutingFormFilter,
+  } = filter;
+  const initialConfigIsReady = !!(initialConfig?.teamId || initialConfig?.userId || initialConfig?.isAll);
+  const [startDate, endDate] = dateRange;
+
+  const { data: headers } = trpc.viewer.insights.routingFormResponsesHeaders.useQuery(
+    {
+      teamId: selectedTeamId,
+      isAll: isAll ?? false,
+      organizationId: initialConfig?.organizationId,
+      routingFormId: selectedRoutingFormId ?? undefined,
+    },
+    {
+      enabled: initialConfigIsReady,
+    }
+  );
+
+  const { data, fetchNextPage, isFetching, hasNextPage } =
+    trpc.viewer.insights.routingFormResponses.useInfiniteQuery(
+      {
+        teamId: selectedTeamId,
+        startDate: startDate.toISOString(),
+        endDate: endDate.toISOString(),
+        userId: selectedMemberUserId ?? undefined,
+        isAll: isAll ?? false,
+        routingFormId: selectedRoutingFormId ?? undefined,
+        bookingStatus: selectedBookingStatus ?? undefined,
+        fieldFilter: selectedRoutingFormFilter ?? undefined,
+        limit: 10,
+      },
+      {
+        getNextPageParam: (lastPage) => lastPage.nextCursor,
+        placeholderData: keepPreviousData,
+        trpc: {
+          context: { skipBatch: true },
+        },
+        enabled: initialConfigIsReady,
+      }
+    );
+
+  const flatData = useMemo(() => data?.pages?.flatMap((page) => page.data) ?? [], [data]);
+  const totalDBRowCount = data?.pages?.[0]?.total ?? 0;
+  const totalFetched = flatData.length;
+
+  const processedData = useMemo(() => {
+    return flatData.map((response) => {
+      const row: RoutingFormTableRow = {
+        id: response.id,
+        formName: response.form.name,
+        formId: response.form.id,
+        createdAt: response.createdAt,
+        routedToBooking: response.routedToBooking,
+      };
+
+      Object.entries(response.response).forEach(([fieldId, field]) => {
+        const header = headers?.find((h) => h.id === fieldId);
+
+        if (header?.options) {
+          if (Array.isArray(field.value)) {
+            // Map the IDs to their corresponding labels for array values
+            const labels = field.value.map((id) => {
+              const option = header.options?.find((opt) => opt.id === id);
+              return option?.label ?? id;
+            });
+            row[fieldId] = labels;
+          } else {
+            // Handle single value case
+            const option = header.options?.find((opt) => opt.id === field.value);
+            row[fieldId] = option?.label ?? field.value;
+          }
+        } else {
+          row[fieldId] = field.value;
+        }
+      });
```

Comment:
Get the field values of the forms that we already fetched from headers otherwise the values inside the form are UUIDs  as thats how they are stored in JSON